src/cargo/core/registry.rs

```diff
@@ -178,13 +178,13 @@ impl<'cfg> PackageRegistry<'cfg> {
 
     pub fn register_lock(&mut self, id: PackageId, deps: Vec<PackageId>) {
         trace!("register_lock: {}", id);
-        for dep in deps.iter() {
+        for dep in &deps {
```

Comment:
I personally prefer `for x in foo.iter()` rather than `for x in &foo`, can this stay the same?

Reply:
Those were accepted before, therefore I kept going.

Personally, I prefer the ampersand-version because `for foo in bar` vs. `for foo in bar.iter()` both both signal "iteration" but don't signal ownership; `for foo in bar` vs. `for foo in &bar` is still both "iteration" and makes the difference in ownership visible.

Reply:
Hm I think I missed those before :(

Can these stay as `.iter()`?

Reply:
I'll reset the branch, remove the `foo.iter() -> &foo` and redo the commit.

Should the `or_insert_with` go as well? The rationale is to avoid creating a new object that usually gets tossed right way (even if  `HashMap::new` doesn't allocate)

Reply:
Something like `or_insert_with` is fine sometimes, but changing `or_insert(Foo::new())` to `or_insert_with(Foo::new)` I think is decreasing readabilty for noa ctual gains.

Reply:
rebased and squashed, no new trivial `or_insert_with` or `&foo` instead of `foo.iter()`